src/libaktualizr/primary/reportqueue.cc

```diff
@@ -32,17 +33,15 @@ void ReportQueue::run() {
 
 void ReportQueue::enqueue(std::unique_ptr<ReportEvent> event) {
   {
-    std::lock_guard<std::mutex> lock(m_);
-    report_queue_.push(std::move(event));
+    //std::lock_guard<std::mutex> lock(m_);
```

Comment:
Is the lock really no longer needed here?

Reply:
Added it back.